packages/apollo-server-core/src/requestPipeline.ts

```diff
@@ -217,7 +217,7 @@ export async function processGraphQLRequest<TContext>(
     } catch (err) {
       logger.warn(
         'An error occurred while attempting to read from the documentStore. ' +
-          (err && err.message) || err,
+          err?.message || err,
```

Comment:
Would you be interested in updating these usages of `||` to the null coalesce `??` operator as well? Keeping in mind the slight functional difference between the two (i.e. if `err?.message === ''`). Possibly a separate PR.
```suggestion
          err?.message ?? err,
```